pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller_test.go

```diff
@@ -342,6 +372,39 @@ func TestReconcileClusterDeployment(t *testing.T) {
 				r.Delete(gomock.Any()).Times(0)
 			},
 		},
+		{
+			name: "Test Hibernation",
+			localObjects: []runtime.Object{
+				hibernatingClusterDeployment(),
+				testSecret(),
+				testSecretRef(),
+				testSyncSet(),
+				testDeadMansSnitchIntegration(),
+			},
+			expectedSyncSets: &SyncSetEntry{
+				name:                     testClusterName + "-" + snitchNamePostFix + "-" + config.RefSecretPostfix,
+				referencedSecretName:     testClusterName + "-" + snitchNamePostFix + "-" + config.RefSecretPostfix,
+				clusterDeploymentRefName: testClusterName,
+			},
+			expectedSecret: &SecretEntry{
+				name:                     testClusterName + "-" + snitchNamePostFix + "-" + config.RefSecretPostfix,
+				snitchURL:                testSnitchURL,
+				clusterDeploymentRefName: testClusterName,
+			},
+			verifySyncSets: verifyNoSyncSet,
+			verifySecret:   verifyNoSecret,
+			setupDMSMock: func(r *mockdms.MockClientMockRecorder) {
+				r.FindSnitchesByName(gomock.Any()).Return([]dmsclient.Snitch{
+					{Token: testSnitchToken},
+				}, nil)
+				r.Delete(gomock.Any()).Return(true, nil)
+
+				r.Create(gomock.Any()).Times(0)
+				r.Update(gomock.Any()).Times(0)
+				r.CheckIn(gomock.Any()).Times(0)
+				r.Delete(gomock.Any()).Times(0)
```

Comment:
```suggestion
```
We do expect deletes, this is not needed?

Reply:
This checks whether any extra calls to delete occur outside the expected. I have done it this was to keep consistency with the other tests but might be worth switching to using `gomock.InOrder` to verify this.